usr/src/uts/armv8/Makefile.rules

```diff
@@ -65,6 +66,65 @@ $(OBJS_DIR)/%.o:		$(UTSBASE)/armv8/io/gfx_private/%.c
 	$(COMPILE.c) -o $@ $<
 	$(CTFCONVERT_O)
 
+#
+# Rules for the direcct-boot shim.
+#
+DBOOT_OBJS_DIR	= dboot/$(OBJS_DIR)
+DBOOT_MACH_64	= -D_BOOT_TARGET_aarch64
+DBOOT_DEFS	= -D_BOOT $(DBOOT_MACH_$(CLASS))
+DBOOT_DEFS	+= -D_MACHDEP -U_KERNEL
+DBOOT_FLAGS	= $(aarch64_CFLAGS) $(CSTD) $(COPTFLAG) $(C_BIGPICFLAGS)
+DBOOT_FLAGS	+= $(CCVERBOSE) $(CERRWARN) $(CCNOAUTOINLINE)
+DBOOT_FLAGS	+= -_gcc=-march=armv8-a+nofp+nosimd
```

Comment:
How does this differ from -mgeneral-regs-only?

Reply:
That's a good question - I'm not sure, and it shouldn't really be necessary anyway given that we should be able to use fp at this point. Our compiler architecture level means we won't get SVE, so I think it should work that way.

Reply:
The Arm docs haver this to say:
> -mgeneral-regs-only: Prevents the use of floating-point instructions or floating-point registers.
> Note
> The  -mgeneral-regs-only option is only valid with the  aarch64-arm-none-eabi target. Use  -mfpu=none to prevent the use of floating-point instructions or floating-point registers for the  armv8a-arm-none-eabi target.

That said, _compiling_ worked file with -mgeneral-regs-only, but linking was unhappy.

`-march=armv8-a+nofp+nosimd` basically turns on only base armv8.0-a profile instructions while explicitly turning off support for floating point and SIMD, which is basically what we'd be trying to accomplish with general registers only.